pkg/volume/iscsi/iscsi_util_test.go

```diff
@@ -275,43 +284,49 @@ func TestClonedIfaceShowError(t *testing.T) {
 	plugin := plugins[0]
 	fakeMounter := iscsiDiskMounter{
 		iscsiDisk: &iscsiDisk{
-			Iface:  "192.168.1.10:pv0001",
+			Iface:  TestIface,
 			plugin: plugin.(*iscsiPlugin)},
 		exec: fakeExec,
 	}
-	cloneIface(fakeMounter)
-	if cmdCount != 1 {
-		t.Errorf("expected 1 CombinedOutput() calls, got %d", cmdCount)
+	err := cloneIface(fakeMounter)
+	if err == nil {
+		t.Errorf("expect to receive error, nil received")
+	}
+	if fakeExec.CommandCalls != len(scripts) {
+		t.Errorf("expected 1 CombinedOutput() calls, got %d", fakeExec.CommandCalls)
 	}
 
 }
 
 func TestClonedIfaceUpdateError(t *testing.T) {
-	cmdCount := 0
-	fakeExec := mount.NewFakeExec(func(cmd string, args ...string) ([]byte, error) {
-		cmdCount++
-		if cmd != "iscsiadm" {
-			t.Errorf("iscsiadm command expected, got %q", cmd)
-		}
-		switch cmdCount {
-		case 1:
-			// iscsiadm -m iface -I <iface> -o show
-			return []byte("iface.ipaddress = <empty>\niface.transport_name = tcp\niface.initiatorname = <empty>\n"), nil
-
-		case 2:
-			// iscsiadm -m iface -I <newIface> -o new
-			return []byte("New interface 192.168.1.10:pv0001 added"), nil
-		case 3:
-			// iscsiadm -m iface -I <newIface> -o update -n <key> -v <val>
-			return []byte(""), nil
-		case 4:
-			return []byte(""), errors.New("test error")
-		case 5:
-			// iscsiadm -m iface -I <newIface> -o delete
-			return []byte(""), nil
-		}
-		return nil, fmt.Errorf("Unexpected exec call nr %d: %s", cmdCount, cmd)
-	})
+	fakeExec := &testingexec.FakeExec{}
+	scripts := []volumetest.CommandScript{
+		{
+			Cmd:    "iscsiadm",
+			Args:   []string{"-m", "iface", "-I", "", "-o", "show"},
+			Output: "iface.ipaddress = <empty>\niface.transport_name = tcp\niface.initiatorname = <empty>\n",
```

Comment:
@msau42: I went ahead and set the `Output` for the `iscsiadm show` command. This causes the code in https://github.com/kubernetes/kubernetes/blob/e7ce8d8a6f82a366a94ee7eb2a29b023cb361007/pkg/volume/iscsi/iscsi_util.go#L837-L855 to be used instead of being skipped, which is a good thing.

However, since these tests are actually trying to verify exact commands, I have to make sure that the commands are run the exact same way every time.  Because https://github.com/kubernetes/kubernetes/blob/e7ce8d8a6f82a366a94ee7eb2a29b023cb361007/pkg/volume/iscsi/iscsi_util.go#L888 is used to iterate over a map, and map iteration is random, this was causing the test to be flaky.

Thus, I added a bit of code in `iscsi_util.go` to make that iteration be stable. Hope that's okay. It's either that or don't validate the commands anymore. I think this is better.